accel/tcg/log_instr.c

```diff
@@ -1036,6 +1065,319 @@ void qemu_log_instr_extra(CPUArchState *env, const char *msg, ...)
+            switch (c) {
+            case 'c':
+                g_string_append_printf(string, bounce_buf, (args++)->charv);
+                format = false;
+                i = 0;
+                break;
+            case 'd':
+            case 'i':
+                if (is_long_long) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->longlongv);
+                } else if (is_long) {
+                    g_string_append_printf(string, bounce_buf, (args++)->longv);
+                } else if (is_short) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->shortv);
+                } else {
+                    g_string_append_printf(string, bounce_buf, (args++)->intv);
+                }
+                format = false;
+                i = 0;
+                break;
+            case 'u':
+            case 'x':
+            case 'X':
+            case 'o':
+                if (is_long_long) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->ulonglongv);
+                } else if (is_long) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->ulongv);
+                } else if (is_short) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->ushortv);
+                } else {
+                    g_string_append_printf(string, bounce_buf, (args++)->uintv);
+                }
+                format = false;
+                i = 0;
+                break;
+            case 'e':
+            case 'E':
+            case 'f':
+            case 'g':
+            case 'G':
+                if (is_long) {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->doublev);
+                } else {
+                    g_string_append_printf(string, bounce_buf,
+                                           (args++)->floatv);
+                }
+                format = false;
+                i = 0;
+                break;
+            case 's':
+            case 'p':
+                g_string_append_printf(string, bounce_buf, (args++)->ptrv);
+                format = false;
+                i = 0;
+                break;
+            case '%': format = false; break;
+            case 'h': is_short = true; break;
+            case 'l':
+                if (is_long)
+                    is_long_long = true;
+                is_long = true;
+                break;
+            default: {
+            }
+            }
+        }
+        assert(i != 256);
+    }
+    if (i != 256) {
+        g_string_append_printf(string, bounce_buf);
+    }
+
+#pragma clang diagnostic pop
+}
+
+TCGv_i64 qemu_log_printf_valid_entries;
+
+void qemu_log_printf_create_globals(void)
```

Comment:
See comment on #148 May wan to move it to qemu_log_instr_init to limit the initialization hooks that we need to add to targets.

Reply:
Now somewhere more generic.